rfcs/text/????-hook-service.md

```diff
@@ -0,0 +1,134 @@
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+TODO:
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+# Prior art
+
+[prior-art]: #prior-art
+
+## `editableRef`
+
+[editableRef]: #editableref
+
+The first exploration of this space was the introduction of `editableRef` in the HASH app, currently powering the rich text blocks.
+
+This was designed to avoid blocks being coupled to the particular rich text editing solution implemented in HASH, and in that it succeed.
+
+`editableRef` is a function provided by the embedding application which a block can call with a DOM node rendered by the block, which they wish the embedding application to render rich text present on the `text` field on the block entity into.
+
+It was also used by the embedding application (HASH) to know that any particular block had an editable text section within it. The particular solution HASH uses for rich text editing needs (or needed) to know at the time blocks are loaded that a user can edit text within them. This worked because a block would use `editableRef` by reading it from the properties passed to it, which means the `editableRef` key would be in the block's schema.
+
+`editableRef` is an unsuitable candidate for standardisation for a number of reasons:
+- It does not indicate which block entity property on which to store the rich text data
+- It means that the type of the value in the `text` field is unknown to the block, and may not match the value in the block's schema
+- It has a one time use – you cannot use it to make multiple editable islands within a block (think a blockquote with a caption)
+- The requirement for the use of the feature to show up in the block's schema can make certain kinds of blocks impossible (i.e, blocks that may or may not have editable sections depending on their current state).
+- An embedding application can only provide one implementation for `editableRef`, which means there may be a mismatch between the implementation provided by the embedding application and the feature expected by the block. I.e, an embedding application may use `editableRef` to provide image editing features, whereas the block may expect it to be used to provide rich text editing features. 
+
+Another minor reason is its naming is tied into React – i.e, "editableRef". 
+
+This proposal fixes all of these problems.
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+- Is hook the best name for this service? Other options include (not exhaustive):
+  - editable/editing/editor service
+  - view service
+  - UI service
+  - plugin service
+  - DOM service
+  - extension service
+- Is the name 'hook' too tied into React and therefore confusing? 
```

Comment:
No, I don't think so. It has a literal metaphorical meaning that could be understood by non-programmers, nevermind non-React developers.

Reply:
Hooks are also generally a computing term (and webhooks are a thing) so I don't think it's React-centric.

Reply:
@nathggns Let's remove this as a question.

Reply:
```suggestion
```